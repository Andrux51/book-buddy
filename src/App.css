.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.bb-description {
  max-width: 720px;
  margin-left: auto;
  margin-right: auto;
}

@keyframes App-logo-spin {
  /* from { transform: rotate(0deg); }
  to { transform: rotate(360deg); } */
}

/* **[attr*=value] Does not work in IE or Edge** */
button[class*="bb-btn-"] {
  border: none;
  text-decoration: none;
  color: white;
  font-size: 17px;
  font-weight: bold;
  border-radius: 3px;
  padding: 10px 0;
  cursor: pointer;
}
button[class*="bb-btn-"] a {
  color: white;
  text-decoration: none;
  padding: 10px 15px;
}

.bb-btn-route {
  background-color:#6c9;
}
button[type="submit"] {
  background-color:#627cf1;
  padding: 10px 15px;
}

.bb-form-input[type="text"] {
  padding:10px;
  border-radius: 3px;
  border: 1px solid #ccc;
  margin-right: 1em;
  font-size:16px;
}