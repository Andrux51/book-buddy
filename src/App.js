import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom'
import './App.css';
import WelcomePage from './components/Welcome.page';
import GetStartedPage from './components/GetStarted.page';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      subscriber: {
        email: '',
        genresChosen: [],
        name: ''
      }
    };
  }

  updateSubscriber = (key, val) => {
    let subscriber = Object.assign({}, this.state.subscriber);

    subscriber[key] = val;

    this.setState({ subscriber });
  }

  render() {
    return (
      <Router>
        <div className="App">
          <header className="App-header">
            <Link to="/">
              <img src="http://placehold.it/80x80?text=logo" className="App-logo" alt="logo" />
            </Link>
            <h1 className="App-title">Welcome to BookBuddy</h1>
          </header>
          <div>
            <Route path="/"
              exact
              component={WelcomePage}
            />
            <Route path="/get-started"
              render={() =>
                <GetStartedPage
                  subscriber={this.state.subscriber}
                  updateSubscriber={this.updateSubscriber}
                />
              }
            />
          </div>
        </div>
      </Router>
    );
  }
}

export default App;
