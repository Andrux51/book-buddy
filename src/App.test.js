import React from 'react';
import App from './App';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';
import WelcomePage from './components/Welcome.page';
import GetStartedPage from './components/GetStarted.page';

let reactRouter, app;

test(`renders without crashing`, () => {
  const component = renderer.create(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

beforeEach(() => {
  reactRouter = mount(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  app = reactRouter.find(App).instance();
});

describe(`Main app component`, () => {
  beforeEach(() => {
  });

  test(`find links`, () => {
      // two ways to write the same property
      expect(reactRouter.find({href: '/'})).toHaveLength(1);
      expect(reactRouter.find(`a[href="/get-started"]`)).toHaveLength(1);
  });

  // testing react-router sucks, but this was more painful than it should have been
  test(`GetStartedPage works...`, () => {
    reactRouter.find({href: '/get-started'}).simulate('click', {button: 0});

    expect(reactRouter.find(`strong`).text()).toBe(`2-4`);
  });

  test(`subscriber object is built`, () => {
    expect(typeof app.state.subscriber.email).toBe('string');
    expect(Array.isArray(app.state.subscriber.genresChosen)).toBe(true);
    expect(typeof app.state.subscriber.name).toBe('string');
  });

  test(`update subscriber`, () => {
    expect(typeof app.state.subscriber).toBe('object');
  });
});
