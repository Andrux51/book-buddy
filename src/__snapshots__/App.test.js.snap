// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="App-logo"
          src="http://placehold.it/80x80?text=logo"
        />
      </a>
      <h1
        className="App-title"
      >
        Welcome to BookBuddy
      </h1>
    </header>
    <div>
      <div>
        <p
          className="App-intro bb-description"
        >
          <strong>
            Upgrade your inbox
          </strong>
           with daily deals on free and discounted ebooks across a multitude of genres. Discover new authors both traditionally and indepedently published.
          <br />
          <br />
          Choose the genres that appeal most to you, and we'll curate an extra choice here and there to help you discover new stuff, too.
        </p>
        <button
          className="bb-btn-route"
          type="button"
        >
          <a
            href="/get-started"
            onClick={[Function]}
          >
            Get Started
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
